File: restaurant_pos/lib/models/order_request.dart
enum RequestType { order, callStaff, payment }
enum RequestStatus { pending, confirmed, cancelled }

class OrderRequest {
  final String id;
  final String tableId;
  final String tableName;
  final RequestType type;
  final RequestStatus status;
  final DateTime createdAt;
  final List<RequestItem>? items;
  final String? note;

  OrderRequest({
    required this.id,
    required this.tableId,
    required this.tableName,
    required this.type,
    this.status = RequestStatus.pending,
    required this.createdAt,
    this.items,
    this.note,
  });

  factory OrderRequest.fromJson(Map<String, dynamic> json) {
    return OrderRequest(
      id: (json['id'] ?? json['Id'])?.toString() ?? '',
      tableId: (json['tableId'] ?? json['TableId'])?.toString() ?? '',
      tableName: (json['tableName'] ?? json['TableName'])?.toString() ?? '',
      type: RequestType.values.firstWhere(
        (e) => e.name == (json['type'] ?? json['Type']), 
        orElse: () => RequestType.order
      ),
      status: RequestStatus.values.firstWhere(
        (e) => e.name == (json['status'] ?? json['Status']), 
        orElse: () => RequestStatus.pending
      ),
      createdAt: DateTime.parse((json['createdAt'] ?? json['CreatedAtUtc'] ?? DateTime.now().toIso8601String())),
      note: json['note'] ?? json['Note'],
      items: (json['items'] ?? json['Items']) != null 
          ? ((json['items'] ?? json['Items']) as List).map((i) => RequestItem.fromJson(i)).toList()
          : null,
    );
  }
}

class RequestItem {
  final String productId;
  final String productName;
  final int quantity;
  final String? note;

  RequestItem({
    required this.productId,
    required this.productName,
    required this.quantity,
    this.note,
  });

  factory RequestItem.fromJson(Map<String, dynamic> json) {
    return RequestItem(
      productId: (json['productId'] ?? json['ProductId'])?.toString() ?? '',
      productName: (json['productName'] ?? json['ProductName'])?.toString() ?? '',
      quantity: (json['quantity'] ?? json['Quantity']) as int? ?? 1,
      note: json['note'] ?? json['Note'],
    );
  }
}

File: restaurant_pos_web/backend/Models/OrderRequest.cs
using System.ComponentModel.DataAnnotations;

namespace RestaurantPos.Api.Models;

public class OrderRequest
{
    [Key]
    public string Id { get; set; } = Guid.NewGuid().ToString("N");
    public string RestaurantId { get; set; } = string.Empty;
    public string TableId { get; set; } = string.Empty;
    public string TableName { get; set; } = string.Empty;
    public string Type { get; set; } = "order"; // order, callStaff, payment
    public string Status { get; set; } = "pending"; // pending, confirmed, cancelled
    public DateTime CreatedAtUtc { get; set; } = DateTime.UtcNow;
    public string? Note { get; set; }

    public List<OrderRequestItem> Items { get; set; } = [];
}

public class OrderRequestItem
{
    [Key]
    public int Id { get; set; }
    public string RequestId { get; set; } = string.Empty;
    public string ProductId { get; set; } = string.Empty;
    public string ProductName { get; set; } = string.Empty;
    public int Quantity { get; set; }
    public string? Note { get; set; }
}
